src/client/components/AddReservations/AddReservations.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { useParams } from "react-router";
+import Meals from "../Meals/Meals";
+import "./AddReservations.css";
+import swal from "sweetalert";
+
+export default function AddReservations(props) {
+  // const [reservation, setReservation] = useState("");
+  const [contactName, setContactName] = useState("");
+  const [numberOfGuests, setNumberOfGuests] = useState(1);
+  const [email, setEmail] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [meal, setMeal] = React.useState("");
+  const params = useParams();
+
+  React.useEffect(() => {
+    const mealUrl = `/api/meals/${params.id}`;
+    fetch(mealUrl)
+      .then((response) => response.json())
+      .then((data) => setMeal(data));
+  }, [params.id]);
```

Comment:
You don't need to have the params.id in the dependency array. An empty array is enough in this case.